java/src/test/java/org/hyperledger/fabric/client/ChaincodeEventsTest.java

```diff
@@ -131,11 +155,140 @@ void sends_valid_request_with_specified_start_block_number_using_sign_bit_for_un
         SignedChaincodeEventsRequest signedRequest = mocker.captureChaincodeEvents();
         ChaincodeEventsRequest request = ChaincodeEventsRequest.parseFrom(signedRequest.getRequest());
 
+        assertChaincodeRequest(request, new ExpectedRequest("NETWORK","CHAINCODE_NAME",Ab.SeekPosition.TypeCase.SPECIFIED, Optional.ofNullable(startBlock),Optional.empty()));
+    }
+
+    @Test
+    void uses_specified_start_block_instead_of_unset_checkpoint() throws Exception {
+        long startBlock = -1;
+        Checkpointer checkpointer = new InMemoryCheckpointer();
+        EventsRequest<ChaincodeEvent> eventsRequest = network.newChaincodeEventsRequest("CHAINCODE_NAME")
+                .startBlock(startBlock)
+                .checkpoint(checkpointer)
+                .build();
+
+        try (CloseableIterator<ChaincodeEvent> iter = eventsRequest.getEvents()) {
+            iter.hasNext(); // Interact with iterator before asserting to ensure async request has been made
+        }
```

Comment:
This gets duplicated a lot. It might be worth having a utility method to do this in a one-line call. Up to you. Not a big deal either way.